var/spack/repos/builtin/packages/hydrogen/package.py

```diff
@@ -0,0 +1,154 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+from spack.spec import UnsupportedCompilerError
+
+
+class Hydrogen(CMakePackage):
```

Comment:
can this be just a separate version of `Elemental` (if all variants are the same)?
Following `xsdk`, you can introduce a convention for versions 

```
version('hydrogen-x.y.z',....)
```

to separate it from the main branch of `elemental`

Reply:
I was struggling with this question myself. The problem is that the hydrogen branch has a reworked cmake environment that will cause the build environment to be very different. Is there a good example where the recipe is very different based on the branch?

Reply:
> Is there a good example where the recipe is very different based on the branch?

do you mean that dependencies are different and the way they are specified/used in cmake is different? if so, i think it's fine to keep it as a separate package.